src/overlays/actors/ovl_Demo_Du/z_demo_du.c

```diff
@@ -1,228 +1,1083 @@
+    DemoDu* this = THIS;
+
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+}
+
+void DemoDu_UpdateEyes(DemoDu* this) {
+    s16* blinkTimer = &this->blinkTimer;
+    s16* eyeTexIndex = &this->eyeTexIndex;
+    s32 pad[3];
+
+    if (DECR(*blinkTimer) == 0) {
+        *blinkTimer = Rand_S16Offset(60, 60);
+    }
+
+    *eyeTexIndex = *blinkTimer;
+    if (*eyeTexIndex >= 3) {
+        *eyeTexIndex = 0;
+    }
+}
+
+void DemoDu_SetEyeTexIndex(DemoDu* this, s16 eyeTexIndex) {
+    this->eyeTexIndex = eyeTexIndex;
+}
+
+void DemoDu_SetMouthTexIndex(DemoDu* this, s16 mouthTexIndex) {
+    this->mouthTexIndex = mouthTexIndex;
+}
+
+// Resets all the values used in the cutscene.
+void DemoDu_Cs02_Reset(DemoDu* this) {
+    this->updateIndex = CS_CS02_SUBSCENE(0);
+    this->drawIndex = 0;
+    this->shadowAlpha = 0;
+    this->demo6KSpawned = 0;
+    this->actor.shape.shadowAlpha = 0;
+    this->unk_1A4 = 0.0f;
+}
+
+void DemoDu_Cs02_CheckIfShouldReset(DemoDu* this, GlobalContext* globalCtx) {
+    static s32 D_8096CE94 = 0;
+
+    if (globalCtx->csCtx.state == 0) {
+        if (D_8096CE94 != 0) {
+            if (this->actor.params == DEMO_DU_TYPE_02) {
+                DemoDu_Cs02_Reset(this);
+            }
+            D_8096CE94 = 0;
+            return;
+        }
+    } else if (D_8096CE94 == 0) {
+        D_8096CE94 = 1;
+    }
+}
+
+// Returns true if the animation finished
+s32 DemoDu_UpdateSkelAnime(DemoDu* this) {
+    return SkelAnime_Update(&this->skelAnime);
+}
+
+void DemoDu_UpdateBgCheckInfo(DemoDu* this, GlobalContext* globalCtx) {
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 75.0f, 30.0f, 30.0f, 5);
+}
+
+CsCmdActorAction* DemoDu_GetNpcAction(GlobalContext* globalCtx, s32 idx) {
+    if (globalCtx->csCtx.state != 0) {
+        return globalCtx->csCtx.npcActions[idx];
+    }
+    return NULL;
+}
+
+s32 DemoDu_IsNpcDoingThisAction(DemoDu* this, GlobalContext* globalCtx, u16 action, s32 idx) {
+    CsCmdActorAction* npcAction;
+
+    npcAction = DemoDu_GetNpcAction(globalCtx, idx);
+    if (npcAction != NULL && npcAction->action == action) {
+        return true;
+    }
+    return false;
+}
+
+s32 DemoDu_IsNpcNotDoingThisAction(DemoDu* this, GlobalContext* globalCtx, u16 action, s32 idx) {
+    CsCmdActorAction* npcAction;
+
+    npcAction = DemoDu_GetNpcAction(globalCtx, idx);
+    if (npcAction != NULL && npcAction->action != action) {
+        return true;
+    }
+    return false;
+}
+
+void DemoDu_MoveToNpcPos(DemoDu* this, GlobalContext* globalCtx, s32 idx) {
+    CsCmdActorAction* npcAction;
+    s16 rotY;
+
+    npcAction = DemoDu_GetNpcAction(globalCtx, idx);
+    if (npcAction != NULL) {
+        this->actor.world.pos.x = npcAction->startPos.x;
+        this->actor.world.pos.y = npcAction->startPos.y;
+        this->actor.world.pos.z = npcAction->startPos.z;
+
+        rotY = npcAction->rot.y;
+        this->actor.shape.rot.y = rotY;
+        this->actor.world.rot.y = rotY;
```

Comment:
looks like a chained assign. Might be able to get rid of `rotY` as a temp.

Reply:
You mean, something like this?
`this->actor.world.rot.y = this->actor.shape.rot.y = npcAction->rot.y;`

Reply:
Yes, exactly.